internal/services/managedapplications/managed_application_resource.go

```diff
@@ -145,35 +146,35 @@ func resourceManagedApplicationCreateUpdate(d *pluginsdk.ResourceData, meta inte
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewApplicationID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	id := applications.NewApplicationID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
 
 	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
+		existing, err := client.Get(ctx, id)
 		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
+			if !response.WasNotFound(existing.HttpResponse) {
 				return fmt.Errorf("failed to check for presence of %s: %+v", id, err)
 			}
 		}
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return tf.ImportAsExistsError("azurerm_managed_application", id.ID())
 		}
 	}
 
-	parameters := managedapplications.Application{
-		Location: utils.String(azure.NormalizeLocation(d.Get("location"))),
-		Kind:     utils.String(d.Get("kind").(string)),
+	parameters := applications.Application{
+		Location: pointer.To(azure.NormalizeLocation(d.Get("location"))),
+		Kind:     d.Get("kind").(string),
 		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
 	if v, ok := d.GetOk("managed_resource_group_name"); ok {
 		targetResourceGroupId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", meta.(*clients.Client).Account.SubscriptionId, v)
```

Comment:
can we switch this to use a `commonid` rather than hard-coding this:

```suggestion
		subscriptionId := meta.(*clients.Client).Account.SubscriptionId
		targetResourceGroupId := commonids.NewResourceGroupID(subscriptionId, v).ID()
```